apps/balancer-tools/src/app/apr/pool/(components)/PoolTokensTable.tsx

```diff
@@ -0,0 +1,116 @@
+          <Table.HeaderCell>Name</Table.HeaderCell>
+          {poolTokensStats[0].weight && (
+            <Table.HeaderCell classNames="w-1/6 text-end p-4">
+              Weight
+            </Table.HeaderCell>
+          )}
+          {poolTokensStats[0].balance && (
+            <Table.HeaderCell classNames="w-1/6 text-end p-4">
+              Balance
+            </Table.HeaderCell>
+          )}
+          {poolTokensStats[0].price && poolTokensStats[0].balance && (
+            <Table.HeaderCell classNames="w-1/6 text-end p-4">
+              Value
+            </Table.HeaderCell>
+          )}
+          {poolTokensStats[0].price && poolTokensStats[0].balance && (
+            <Table.HeaderCell classNames="w-1/6 text-end p-4">
+              Token %
+            </Table.HeaderCell>
+          )}
+        </Table.HeaderRow>
+        <Table.Body>
+          {poolTokensStats.map((token) => (
+            <Table.BodyRow classNames="hover:bg-blue4 hover:cursor-pointer duration-500">
+              <Table.BodyCellLink
+                href={`https://etherscan.io/address/${token.address}`}
+                tdClassNames="w-max"
+                linkClassNames="flex gap-2 items-center"
+              >
+                <div className="w-[25px]">
+                  <Image
+                    src={token.logo}
+                    height={25}
+                    width={25}
+                    className="rounded-full"
+                    alt={`Logo for ${token.symbol}`}
+                  />
+                </div>
+                <div>{token.symbol}</div>
+                <div>
+                  <ExternalLinkIcon />
+                </div>
+              </Table.BodyCellLink>
+              {token.weight && (
+                <Table.BodyCellLink
+                  href={`https://etherscan.io/address/${token.address}`}
+                  tdClassNames="w-6"
+                  linkClassNames="justify-end w-full"
+                >{`${(
+                  parseFloat(token.weight) * 100
+                ).toFixed()}%`}</Table.BodyCellLink>
+              )}
+              {poolTokensStats[0].balance && (
+                <Table.BodyCellLink
+                  href={`https://etherscan.io/address/${token.address}`}
+                  tdClassNames="w-6"
+                  linkClassNames="justify-end w-full"
+                >
+                  {String(tokenBalanceUSD(token.balance)).length <= 3
+                    ? parseFloat(token.balance).toFixed(3)
+                    : tokenBalanceUSD(token.balance)}
+                </Table.BodyCellLink>
+              )}
+              {token.price && poolTokensStats[0].balance && (
+                <Table.BodyCellLink
+                  href={`https://etherscan.io/address/${token.address}`}
+                  tdClassNames="w-6"
+                  linkClassNames="justify-end w-full"
+                >
+                  {`$${(
+                    token.price! * parseFloat(token.balance)
```

Comment:
why use the non null assertion (!) here?

Reply:
There's a `if token.price && poolTokensStats[0].balance`, if that code gets reached should mean that it isn't undefined